src/librustc/traits/project.rs

```diff
@@ -1320,23 +1320,10 @@ fn assoc_ty_def<'cx, 'gcx, 'tcx>(
     let trait_def_id = selcx.tcx().impl_trait_ref(impl_def_id).unwrap().def_id;
     let trait_def = selcx.tcx().trait_def(trait_def_id);
 
-    if !trait_def.is_complete(selcx.tcx()) {
```

Comment:
See #32205 and [the docs for `ExactMatch`](https://github.com/rust-lang/rust/blob/9890e0466d951d1b0020ae38e2a4c0d8c43cfea1/src/librustc/traits/project.rs#L41-L69) (now controlled by `!is_complete`).